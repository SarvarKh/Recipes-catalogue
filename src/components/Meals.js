import { Link } from "react-router-dom";

const Meals = ({meals, clickOnDetailMeal}) => {
    return (
        <>
            <h1 className="titles t-meal">Meals</h1>
            <div className="flex meal-container">
                {
                meals !== undefined ?
                    meals.map(meal => (
                    <Link to="./components/DetailMeal" key={meal.idMeal} className="meal" onClick={() => clickOnDetailMeal(meal.idMeal)}>
                        <div className="img-container">
                            <img src={meal.strMealThumb} />
                        </div>
                        <h3>{meal.strMeal}</h3>
                    </Link>
                    ))
                : <h2>Loading...</h2>
                }
            </div>
        </>
    )
}

export default Meals;
