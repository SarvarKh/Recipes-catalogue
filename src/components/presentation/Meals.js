import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Meals = ({ meals, clickOnDetailMeal }) => (
  <>
    <h1 className="titles t-meal">Meals</h1>
    <div className="flex meal-container">
      {
                meals !== undefined
                  ? meals.map((meal) => (
                    <Link to="./components/DetailMeal" key={meal.idMeal} className="meal" onClick={() => clickOnDetailMeal(meal.idMeal)}>
                      <div className="img-container">
                        <img src={meal.strMealThumb} alt="Meal" />
                      </div>
                      <h3>{meal.strMeal}</h3>
                    </Link>
                  ))
                  : <h2>Loading...</h2>
                }
    </div>
  </>
);

export default Meals;

Meals.propTypes = {
  meals: PropTypes.instanceOf(Array).isRequired,
  clickOnDetailMeal: PropTypes.func.isRequired,
};
