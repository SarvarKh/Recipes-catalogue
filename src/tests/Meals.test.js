import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Meals from '../components/presentation/Meals';

const meals = [
  {
    strMeal: 'Apam balik',
    strMealThumb: 'https://www.themealdb.com/images/media/meals/adxcbq1619787919.jpg',
    idMeal: '53049'
  },
  {
    strMeal: 'Apple & Blackberry Crumble',
    strMealThumb: 'https://www.themealdb.com/images/media/meals/xvsurr1511719182.jpg',
    idMeal: '52893'
  },
];

const clickOnDetailMeal = () => (
    console.log('Test Meals and logging Meals')
)

it('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><Meals meals={meals} clickOnDetailMeal={clickOnDetailMeal} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});