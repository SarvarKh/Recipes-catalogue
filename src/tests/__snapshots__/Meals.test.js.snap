// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    className="titles t-meal"
  >
    Meals
  </h1>,
  <div
    className="flex meal-container"
  >
    <a
      className="meal"
      href="./components/DetailMeal#53049"
      onClick={[Function]}
    >
      <div
        className="img-container"
      >
        <img
          alt="Meal"
          src="https://www.themealdb.com/images/media/meals/adxcbq1619787919.jpg"
        />
      </div>
      <h3>
        Apam balik
      </h3>
    </a>
    <a
      className="meal"
      href="./components/DetailMeal#52893"
      onClick={[Function]}
    >
      <div
        className="img-container"
      >
        <img
          alt="Meal"
          src="https://www.themealdb.com/images/media/meals/xvsurr1511719182.jpg"
        />
      </div>
      <h3>
        Apple & Blackberry Crumble
      </h3>
    </a>
  </div>,
]
`;
